### PR TITLE
fix(shorebird_cli): get MS auth token, add token provider env var for CI

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds `SHOREBIRD_TOKEN_PROVIDER` environment variable requirement for CI login. This lets us know which auth provider we should use to obtain a JWT.

This also adds the `offline_access` scope to our Azure scopes to allow us to obtain a refresh token.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by eseidel** [comment]
Could we just embed this in the token?

**Comment by felangel** [comment]
> Could we just embed this in the token?

Yeah I agree, we could base64 encode the provider + refresh token 🤷

**Comment by bryanoltman** [comment]
Created a CiToken type whose base64-representation will be the SHOREBIRD_TOKEN env var

File: packages/shorebird_cli/lib/src/auth/auth.dart [hunk 1]
```diff
@@ -248,9 +263,15 @@ class Auth {
...
 
   void _loadCredentials() {
-    final token = platform.environment['SHOREBIRD_TOKEN'];
-    if (token != null) {
-      _token = token;
+    final base64Token = platform.environment[shorebirdTokenEnvVar];
+    if (base64Token != null) {
+      try {
+        _token = CiToken.fromBase64(base64Token);
+      } catch (_) {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/auth/auth.dart:270]:
```suggestion
      } catch (_) {
      // TODO(bryanoltman): Remove this legacy behavior after July 2024 or next major release.
      logger.warn('Token failed to parse, assuming an older Google token and re-trying.');
      _token = CiToken(base64Token, googleAuth);
```

Comment (eseidel) [reply packages/shorebird_cli/lib/src/auth/auth.dart:270]: Not sure if it's worth it, but we could do something like this. 🤷

**Review by eseidel — APPROVED** [review]
Consider not forcing existing customers using CI to re-login.  otherwise lgtm.

File: packages/shorebird_cli/lib/src/auth/auth.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/auth/auth.dart:270]: Good call, done. Added more context to the message, along with what steps the user could take to resolve the warning.

**Pull request merged**